assembly/src/parsers/ast/nodes.rs

```diff
@@ -241,3 +242,461 @@ pub enum Instruction {
     CallLocal(u32),
     CallImported(String),
 }
+
+pub trait Serializable: Sized {
+    fn to_bytes(&self) -> Vec<u8>;
+}
+
+impl Serializable for Vec<Node> {
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut vec_bytes: Vec<u8> = Vec::new();
+
+        for node in self {
+            vec_bytes.append(&mut node.to_bytes());
+        }
+        let mut res: Vec<u8> = Vec::new();
+        res.append(&mut (vec_bytes.len() as u64).to_le_bytes().to_vec());
+        res.append(&mut vec_bytes);
+        res
+    }
+}
+
+impl Serializable for Node {
+    fn to_bytes(&self) -> Vec<u8> {
+        match self {
+            Self::Instruction(i) => {
+                let mut instruction_bytes: Vec<u8> = Vec::new();
+                instruction_bytes.push(0); // Instruction id = 0
+                instruction_bytes.append(&mut i.to_bytes());
+                instruction_bytes
+            }
+            Self::IfElse(left, right) => {
+                let mut if_else_bytes: Vec<u8> = Vec::new();
+                if_else_bytes.push(1); // IfElse id = 1
+
+                let mut left_bytes = left.to_bytes();
+                if_else_bytes.append(&mut left_bytes);
+
+                let mut right_bytes = right.to_bytes();
+                if_else_bytes.append(&mut right_bytes);
+
+                if_else_bytes
+            }
+            Self::Repeat(times, nodes) => {
+                let mut repeat_bytes: Vec<u8> = Vec::new();
+                repeat_bytes.push(2); // Repeat id = 2
+
+                repeat_bytes.append(&mut (*times as u64).to_le_bytes().to_vec());
+
+                let mut nodes_bytes = nodes.to_bytes();
+                repeat_bytes.append(&mut nodes_bytes);
+
+                repeat_bytes
```

Comment:
Similar comment as for if-else. Also, `times` could probably be just u32 (not u64).